enterprise/backend/src/metabase_enterprise/transforms_python/python_runner.clj

```diff
@@ -183,16 +184,21 @@
 
         payload                  {:code                code
                                   :library             (t2/select-fn->fn :path :source :model/PythonLibrary)
-                                  :timeout             (transforms-python.settings/python-runner-timeout-seconds)
-                                  :request_id          run-id
+                                  :timeout             (or timeout-secs (transforms-python.settings/python-runner-timeout-seconds))
+                                  :request_id          (or request-id run-id)
                                   :output_url          (:url output)
                                   :output_manifest_url (:url output-manifest)
                                   :events_url          (:url events)
                                   :table_mapping       table-name->url
                                   :manifest_mapping    table-name->manifest-url}
 
-        response                 (transforms.instrumentation/with-python-api-timing [run-id]
-                                   (python-runner-request server-url :post "/execute" {:body (json/encode payload)}))]
+        request-fn               #(python-runner-request server-url :post "/execute" {:body (json/encode payload)})
+
+        response                 (if run-id
```

Comment:
this is a hangover, instrumentation will accept nil runs (see also #64358 ). Will clean up